modules/ding_wayf_dk/modules/ding_wayf_dk_create_user/ding_wayf_dk_create_user.module

```diff
@@ -0,0 +1,397 @@
+    'description' => 'Create new user using WAYF login.',
+    'page callback' => 'ding_wayf_dk_create_user_info_page',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  $items['wayf/create-user/notice'] = array(
+    'title' => 'User creation notice',
+    'description' => 'Notice about account creation.',
+    'page callback' => 'ding_wayf_dk_create_user_notice',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  $items['wayf/create-user/form'] = array(
+    'title' => 'User creation form',
+    'description' => 'Form used to account creation.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_wayf_dk_create_user_form'),
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_wayf_dk_create_user_ctools_plugin_directory($owner, $plugin_type) {
+  return 'plugins/' . $plugin_type;
+}
+
+/**
+ * Implements hook_wayf_dk_login_authenticate_result_alter().
+ *
+ * Store the schacPersonalUniqueID in session to use later when send user create
+ * to the provider.
+ */
+function ding_wayf_dk_create_user_wayf_dk_login_authenticate_result_alter($result) {
+  if (isset($_SESSION['ding_wayf_dk_login_destination']) && $_SESSION['ding_wayf_dk_login_destination'] == 'wayf/create-user/form') {
+    // Extract unique id form the WAYF attributes (the CPR number).
+    preg_match('/\d{10}$/', reset($result['attributes']['schacPersonalUniqueID']), $unique_id);
+    $unique_id = reset($unique_id);
+
+    // Check if age limit should be set.
+    $limit = variable_get('ding_wayf_user_create_age_limit', '');
+    if (!empty($limit) && is_numeric($limit)) {
+
+      // Magic function to get the 1900, 1800 or 2000 prefixes for year.
+      // @see http://da.wikipedia.org/wiki/CPR-nummer
+      $year = substr($unique_id, 4, 2);
+      switch (substr($unique_id, 6, 1)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+          $year = '19' . $year;
+          break;
+
+        case 4:
+        case 9:
+          $year = $year > 36 ? '19' . $year : '20' . $year;
+          break;
+
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+          $year = $year > 57 ? '18' . $year : '20' . $year;
+          break;
+      }
+
+      // Create date time object to use it's diff method.
+      $from = new DateTime();
+      $from->setDate($year, substr($unique_id, 2, 2), substr($unique_id, 0, 2));
+      $to = new DateTime('tomorrow');
+
+      // Check if limit have been reached.
+      if ($from->diff($to)->y < $limit) {
+        // The age limit has to kick in.
+        drupal_set_message('You are not allowed to automatically be created because of the age limit. Please go to your local library to registry.');
+
+        // Redirect to front page.
+        drupal_goto('<front>');
+      }
+    }
+
+    $_SESSION['ding_wayf_dk_create_user'] = $unique_id;
```

Comment:
`ding_wayf_dk_create_user` is an odd name for a value which contains the id for a user. How about `ding_wayf_dk_create_user_id`,  `ding_wayf_dk_create_user_cpr` or the like?